### PR TITLE
Feature/tab-completion

### PR DESCRIPTION
**Pull request by kema91**

### CODE REVIEW

File: src/main/java/de/retest/recheck/cli/subcommands/Completion.java [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	@Spec
+	protected CommandSpec spec;
+
+	@Override
+	public void run() {
+		CommandLine root = spec.commandLine();
+		while ( root.getParent() != null ) {
+			root = root.getParent();
+		}
+		final String script = AutoComplete.bash( root.getCommandName(), root );
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/cli/subcommands/Completion.java]: Does `bash` mean it is only Bash-specific? Or is it a file that includes both, Bash and ZSH?

File: src/main/java/de/retest/recheck/cli/subcommands/Completion.java [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+		if ( script.isEmpty() ) {
+			logger.info( "Failed to generate the auto completion script." );
```
Comment (beatngu13) [inline src/main/java/de/retest/recheck/cli/subcommands/Completion.java]: Should be `ERROR` level.

File: README.md [hunk 3]
```diff
@@ -15,18 +15,23 @@ Command-line interface for recheck.
...
+  ignore       Ignore given differences.
+  completion   Generate and display auto completion script.
 ```
 
 
 ## Download and Installation
 
 The recheck CLI is still under development, but we will soon release a minimal working solution.
 
+## Command Line Completion
```
Comment (beatngu13) [inline README.md]: ## Auto-Completion for Bash and ZSH

You can obtain an auto-completion script for Bash and ZSH via the `completion` command. Simply add the resulting output to your `.bash_profile` or `.bashrc`, for example:

```bash
$ echo "source <(recheck completion)" >> ~/.bashrc
```

Comment (modulo11) [inline README.md]: I'd name this **Enabling shell autocompletion**.

File: src/main/java/de/retest/recheck/cli/subcommands/Completion.java (hunk 1 above)
Comment (kema91) [reply src/main/java/de/retest/recheck/cli/subcommands/Completion.java]: I think AutoComplete.bash includes both, bash and ZSH. AutoComplete also provides no special method for ZSH.

File: README.md [hunk 4]
```diff
@@ -15,18 +15,27 @@ Command-line interface for recheck.
...
+  ignore       Ignore given differences.
+  completion   Generate and display auto completion script.
 ```
 
 
 ## Download and Installation
 
 The recheck CLI is still under development, but we will soon release a minimal working solution.
 
+## Enabling shell autocompletion
```
Comment (beatngu13) [inline README.md]: Grammar Nazi: we use title-style capitalization (see "Download and Installation"), hence, it should be "Enabling Shell Auto-Completion".

**Pull request merged**